### PR TITLE
x/lock: Fix `ExtendLockup` API 

### PR DESCRIPTION
**Pull request by mattverse**
Sub-component of: #1838

## What is the purpose of the change

The current `ExtendLockup` API directly takes in the lock object as a pointer as its parameter. This PR changes this existing API to take in lock ID to improve safety for this API

## Brief Changelog

Improve`ExtendLockup` API safety.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not documented

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]
We should have a changelog for this IMO

File: x/lockup/keeper/lock.go [hunk 1]
```diff
@@ -384,8 +393,13 @@ func (k Keeper) ExtendLockup(ctx sdk.Context, lock types.PeriodLock, newDuration
...
-	oldLock := lock
+	// completely delete existing lock refs
+	err = k.deleteLockRefs(ctx, unlockingPrefix(lock.IsUnlocking()), *lock)
+	if err != nil {
+		return err
+	}
 
+	oldDuration := lock.GetDuration()
 	if newDuration != 0 {
 		if newDuration <= lock.Duration {
```
Comment (ValarDragon) [inline x/lockup/keeper/lock.go:404]:
```suggestion
		if newDuration <= oldDuration {
```
If we're defining it right above =p

File: x/lockup/keeper/lock.go [hunk 2]
```diff
@@ -400,25 +414,19 @@ func (k Keeper) ExtendLockup(ctx sdk.Context, lock types.PeriodLock, newDuration
...
 	}
 
-	// update lockup
-	err := k.deleteLockRefs(ctx, unlockingPrefix(oldLock.IsUnlocking()), oldLock)
-	if err != nil {
-		return err
-	}
-
-	err = k.addLockRefs(ctx, lock)
+	err = k.addLockRefs(ctx, *lock)
```
Comment (ValarDragon) [inline x/lockup/keeper/lock.go:417]:
```suggestion
	// add lock refs with the new duration
	err = k.addLockRefs(ctx, *lock)
```

**Review by ValarDragon — APPROVED** [review]
LGTM, but lets add a Go API changelog entry

**Comment by mattverse** [comment]
thanks for adding the chnage log and merging!

**Pull request merged**